modules/cassandra/src/main/java/org/testcontainers/cassandra/CassandraContainer.java

```diff
@@ -0,0 +1,177 @@
+package org.testcontainers.cassandra;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.testcontainers.cassandra.delegate.CassandraDatabaseDelegate;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.ext.ScriptUtils;
+import org.testcontainers.ext.ScriptUtils.ScriptLoadException;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.File;
+import java.net.InetSocketAddress;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+
+/**
+ * Testcontainers implementation for Apache Cassandra.
+ * <p>
+ * Supported image: {@code cassandra}
+ * <p>
+ * Exposed ports: 9042
+ */
+public class CassandraContainer<SELF extends CassandraContainer<SELF>> extends GenericContainer<SELF> {
```

Comment:
```suggestion
public class CassandraContainer extends GenericContainer<CassandraContainer> {
```

Reply:
Thank you @kiview for the explanation. I'd just copied this from the previous implementation. I will adapt this as suggested.